openmdao/core/group.py

```diff
@@ -1419,9 +1420,20 @@ def _setup_connections(self):
                     if np.prod(src_indices.shape) == 0:
                         continue
 
+                    if is_slice and src_indices.size == np.prod(in_shape):
+                        flat_array_slice_check = True
+                    else:
+                        flat_array_slice_check = False
+
+                    if any('flat_src_indices' in subsys._var_abs2meta[name]
+                           for name in subsys._var_abs2meta):
+                        msg = ("%s: flat_src_indices has no effect when using om_slicer to "
+                               "slice array." % (self.msginfo))
+                        simple_warning(msg)
+
                     # initial dimensions of indices shape must be same shape as target
                     for idx_d, inp_d in zip(src_indices.shape, in_shape):
-                        if idx_d != inp_d:
+                        if idx_d != inp_d and not flat_array_slice_check:
```

Comment:
If `flat_array_slice_check` never changes in the loop, can't you pull the check outside the loop (all of these loops)? Since you aren't checking, you could just do a `continue` before that whole block of checks to move to the next index in the outer loop.

Reply:
Also, is the flag definition backwards? We check when `and *not* flat_array_slice_check` is True.  Sounds like when check is True, we don't check, unless i am missing something.

Reply:
We're checking to see if it's a slice and if the sizes match and if it's False, we skip the check